server/routerlicious/packages/routerlicious-base/src/utils/sessionHelper.ts

```diff
@@ -134,20 +136,50 @@ async function updateExistingSession(
         isSessionActive: false,
     };
     try {
-        await documentsCollection.upsert(
+        const result = await documentsCollection.findAndUpdate(
             {
                 documentId,
+                "session.isSessionAlive": false,
```

Comment:
Would it be also needed to pass the tenantId here? Even though it is very unlikely using GUIDs, the same documentId could exist for 2 different tenants. Or, if another consumer of the open source packages implements a different way of defining documentIds (e.g. client-provided docIds), then could this also be a problem?

Reply:
Got it. Thanks. :) I Added the tenant id here to be safer.